update debugging explainer with debugging data (#157)
privacysandbox/protected-auction-services-docs: debugging_protected_audience_api_services.md

    partitioned by destination:** Similar to the above, but partitioned by the
    request destination.

## Debugging data

> Bidding and Auction servers can be built in 2 [build_flavor](https://github.com/privacysandbox/bidding-auction-servers/blob/e40a4fccdce168379189ab7b6b87b55b1e3f736d/BUILD#L163), `prod` and `non_prod`, only `prod` image can pass attestation and decrypt

Reviewer: ```suggestion
Bidding and Auction servers can be built in 2 [build_flavor](https://github.com/privacysandbox/bidding-auction-servers/blob/e40a4fccdce168379189ab7b6b87b55b1e3f736d/BUILD#L163), `prod` and `non_prod`. Only `prod` images can pass attestation and work in production (i.e. decrypt live traffic requests).
```

production request, but `non_prod` have more debugging methods with requests with fake encryption.

> 

Reviewer: ```suggestion

Debugging data consists of the following:
```

1. Verbose logs

    This is the verbose logs from the B&A code.
    - Export channel

>       - In `prod`, they are only exported through Open Telemetry log, which can be viewed in cloud log explorer.

Reviewer: ```suggestion
      - In `prod`, verbose logs are exported through [Open Telemetry log](https://opentelemetry.io/docs/concepts/signals/logs/), which can be viewed in cloud log explorer.
```

      - In `non_prod`, they are also printed in stderr.

    - Log types
      - System logs:

>         These logs are not related to user device request, they are always exported in both `prod` and `non_prod`.

Reviewer: ```suggestion
        These logs are not related to user device request, they are always exported in both `prod` and `non_prod` builds.
```

      - Request logs:

>         These logs are the result of server processing encrypted request. With `prod`, they are only exported for [AdTech consented debugging](#adtech-consented-debugging). With `non_prod`, they are always exported.

Reviewer: ```suggestion
        These logs are the result of server processing encrypted request. In `prod` builds, these are only exported for [AdTech consented debugging](#adtech-consented-debugging) requests. In `non_prod` builds, they are always exported.
```

>     Each server has `PS_VERBOSITY` flag, which sets the verbosity level. High `PS_VERBOSITY` can slow down the `non_prod` server, because there can be large volume of request logs at high QPS, `PS_VERBOSITY` should be `<=3` in this case.

Reviewer: ```suggestion
    Each server has `PS_VERBOSITY` flag, which sets the verbosity level of the output logs. A higher verbosity will give more logging data, but can also slow down request processing. This is especially relevant in `non_prod` builds because there can be large volume of request logs. At high QPS, we recommend setting `PS_VERBOSITY`   `<=3`.
```

Debugging data consists of the following:
1. Verbose logs

    This is the verbose logs from the B&A code.
    - Export channel
      - In `prod`, verbose logs are exported through [Open Telemetry log](https://opentelemetry.io/docs/concepts/signals/logs/), which can be viewed in cloud log explorer.
      - In `non_prod`, they are also printed in stderr.

    - Log types
      - System logs:
        These logs are not related to user device request, they are always exported in both `prod` and `non_prod` builds.
      - Request logs:
        These logs are the result of server processing encrypted request. In `prod` builds, these are only exported for [AdTech consented debugging](#adtech-consented-debugging) requests. In `non_prod` builds, they are always exported.

    Each server has `PS_VERBOSITY` flag, which sets the verbosity level of the output logs. A higher verbosity will give more logging data, but can also slow down request processing. This is especially relevant in `non_prod` builds because there can be large volume of request logs. At high QPS, we recommend setting `PS_VERBOSITY`   `<=3`.

> 2. Event message

Reviewer: Should we call this "Request/Response export"?

Author: There can be others in the Event message, like udf log and kv signal.

>     This is a [structured proto message](https://github.com/privacysandbox/bidding-auction-servers/blob/e40a4fccdce168379189ab7b6b87b55b1e3f736d/api/bidding_auction_servers.proto#L1582) to include the API proto object.

Reviewer: ```suggestion
    Request and response from the server can also be exported in a structured format. The following  [structured proto message](https://github.com/privacysandbox/bidding-auction-servers/blob/e40a4fccdce168379189ab7b6b87b55b1e3f736d/api/bidding_auction_servers.proto#L1582) defines the proto for these.
```

    It is written to cloud storage through Open Telemetry log collector([example config](https://github.com/privacysandbox/bidding-auction-servers/blob/main/production/deploy/gcp/terraform/services/autoscaling/collector_startup.tftpl)) for [AdTech consented debugging](#adtech-consented-debugging).

> 3. Adtech UDF Log

Reviewer: We shoul dmention which UDFs in the text below.

Author: This should include all the UDFs

>     This is the log from javascript `console.log`. They are only exported for [AdTech consented debugging](#adtech-consented-debugging), through both Request logs and Event message.

Reviewer: What is Request log in this context? Is it system log?

Author: it is  mentioned above under "Verbose logs" - Log types - Request logs
